hack/util.sh

```diff
@@ -44,50 +75,132 @@ nodes:
 EOF
 
   kind create cluster --config="${KIND_CONFIG}" --name="${HOST_CLUSTER_NAME}" --kubeconfig="${KUBECONFIG_PATH}"
-  kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="kind-${HOST_CLUSTER_NAME}" create -f "${MANIFEST_DIR}"
-  kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="kind-${HOST_CLUSTER_NAME}" create -f "${CONFIG_DIR}"
-
-	NOFED=kubeadmiral.io/no-federated-resource=1
-	kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="kind-${HOST_CLUSTER_NAME}" annotate deploy -A --all $NOFED
-	kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="kind-${HOST_CLUSTER_NAME}" annotate daemonset -A --all $NOFED
-	kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="kind-${HOST_CLUSTER_NAME}" annotate cm -A --all $NOFED
-	kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="kind-${HOST_CLUSTER_NAME}" annotate secret -A --all $NOFED
-	kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="kind-${HOST_CLUSTER_NAME}" annotate role -A --all $NOFED
-	kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="kind-${HOST_CLUSTER_NAME}" annotate rolebinding -A --all $NOFED
-	kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="kind-${HOST_CLUSTER_NAME}" annotate clusterrole -A --all $NOFED
-	kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="kind-${HOST_CLUSTER_NAME}" annotate clusterrolebinding -A --all $NOFED
-	kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="kind-${HOST_CLUSTER_NAME}" annotate svc -A --all $NOFED
 
   rm "${KIND_CONFIG}"
 }
 
+function util::create_host_kwok_cluster() {
+  local HOST_CLUSTER_NAME=${1}
+  local KUBECONFIG_PATH=${2}
+
+  local KWOK_CONFIG
+  KWOK_CONFIG=$(mktemp)
+  APISERVER_PORT=$(shuf -i 30000-40000 -n 1)
```

Comment:
Why did we not choose to use `$RANDOM`?

Reply:
good question 🙂, i shall switch to using `$RANDOM` for portability